File: Arcade/Intro/C++/16_areSimilar.cpp
bool areSimilar(std::vector<int> a, std::vector<int> b) {
    int errors=0;
    bool canFlip=false;
    int shouldBeA=0;
    int shouldBeB=0;
    for(int i=0;i<a.size();i++){
        if(a[i]!=b[i]){
            errors++;
            if(errors==1) {shouldBeA=b[i]; shouldBeB=a[i];}
            if(errors==2) canFlip = a[i]==shouldBeA && b[i]==shouldBeB;
        }
    }
    if(errors==2){
        return canFlip;
    } else {
        return errors==0;
    }
}

File: Arcade/Intro/JavaScript/16_areSimilar.js
function areSimilar(a, b) {
    
    var firstIndex = -1, 
        diff = 0;
    
    for (var i = 0; i < a.length; i++) {
        if (a[i] !== b[i]) {
            
            if (diff === 2) { return false; }
            diff++;
            
            if (firstIndex === -1) {
                firstIndex = i;
            } else if (a[firstIndex] !== b[i] || b[firstIndex] !== a[i]) {
                return false;
            }
        }
    } return true;
}